dnf/package.py

```diff
@@ -263,15 +263,23 @@ def localPkg(self):
     def remote_location(self, schemes=('http', 'ftp', 'file', 'https')):
         # :api
         """
-        The location from where the package can be downloaded from
+        The location from where the package can be downloaded from. Returns None installed packages.
+        For packages from commandline repository it returns original path or path to the cached
+        file.
 
         :param schemes: list of allowed protocols. Default is ('http', 'ftp', 'file', 'https')
         :return: location (string) or None
         """
+        if self._from_system:
+            return None
+        if self._from_cmdline:
+            if 'file' in schemes:
+                return self.location
```

Comment:
Interestingly if the path is specified with `file:/` protocol the returned location is into dnf cache even if there is no file, if the path is specified without the protocol it returns the actual file location. It is shown in the first two tests: https://github.com/rpm-software-management/ci-dnf-stack/pull/928

Reply:
I know that, but if it return an adres where is no package the it is an issue. The question is why the package is not there. In case that it was there but removed after the command finished or it was never there. I suggest the first option, but I will investigate it.

Reply:
The package is not there because it was removed after command exited. I tried dnf repoquery --location file:///home/jmracek/Projects/libsolv-0.7.10-1.fc31.src.rpm libsolv-0.7.10-1.fc31.src.rpm --setopt=keepcache=true, and the file was preserved.
What about to return None also for commandline packages?

Reply:
@kontura What do you think?

Reply:
I think returning None is fine. 

To be honestly it seems weird to me we can't return a path to a package when the path is part of the input but looking at the code I understand its not that easy and I believe its not worth any bigger changes.

Reply:
I provided a fixup with proposed change.